php-frontend/src/main/java/org/sonar/php/tree/impl/expression/AbstractArrayAssignmentPatternTreeImpl.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.tree.impl.expression;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.php.parser.TreeFactory.Tuple;
+import org.sonar.php.tree.impl.PHPTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.ArrayAssignmentPatternElementTree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+
+public abstract class AbstractArrayAssignmentPatternTreeImpl extends PHPTree {
+
+  private final List<Tree> elementsAndSeparators = new ArrayList<>();
+  private final List<Optional<ArrayAssignmentPatternElementTree>> elements = new ArrayList<>();
+  private final List<SyntaxToken> separators = new ArrayList<>();
+
+  public AbstractArrayAssignmentPatternTreeImpl(@Nullable ArrayAssignmentPatternElementTree firstElement, List<Tuple<SyntaxToken,Optional<ArrayAssignmentPatternElementTree>>> rest) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 182 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)